test/udp_tracker.cpp

```diff
@@ -131,6 +131,8 @@ struct udp_tracker
 				break;
 		}
 
+		if (m_ios.stopped()) return;
+
```

Comment:
imagine the main thread calling stop() right here. immediately after this thread checked it.